### PR TITLE
Sandbox name resonance between sidebar and header

### PR DESCRIPTION
**Pull request by samrith-s**
Update `CollectionInfo` in `Header`'s render function.

**What kind of change does this PR introduce?**
This issue fixes a bug and maintains resonance of the sandbox name between the sidebar and the header.

**What is the current behavior?**
https://github.com/CompuIves/codesandbox-client/issues/1680

**Checklist**:
- [ ] Documentation (*NA*)
- [ ] Tests (*NA*)
- [x] Ready to be merged
- [x] Added myself to contributors table 

The issue was `CollectionInfo` was being passed a reference to a sub-sect of observable `store`, from `Header`. But MobX states that for the observable to trigger a reaction, it needs to be called in `render`. 

The quick fix is to pass a new reference of `sandbox` to`CollectionInfo`, rather than passing the observable.

### CODE REVIEW

**Comment by CompuIves** [comment]
Great! Thanks a lot @samrith-s

File: packages/app/src/app/pages/Sandbox/Editor/Header/index.tsx [hunk 1]
```diff
@@ -154,7 +154,10 @@ const Header = ({ store, signals, zenMode }: Props) => {
 
       {sandbox.owned && (
         <Centered style={{ margin: '0 3rem' }}>
-          <CollectionInfo isLoggedIn={store.isLoggedIn} sandbox={sandbox} />
+          <CollectionInfo
+            isLoggedIn={store.isLoggedIn}
+            sandbox={{ ...sandbox }}
```
Comment (CompuIves) [inline packages/app/src/app/pages/Sandbox/Editor/Header/index.tsx]: Could you leave a comment as to why you do this? Also, maybe `sandbox.toJS()` works too.

Comment (samrith-s) [reply packages/app/src/app/pages/Sandbox/Editor/Header/index.tsx]: Will check that out, and shall leave a comment too!

**Comment by samrith-s** [comment]
@CompuIves done! Do have a look.

**Comment by CompuIves** [comment]
Great! Thank you a lot, really nice to have these contributions! Merging now.

**Pull request merged**